db/crud.go

```diff
@@ -42,14 +42,18 @@ func realDocID(docid string) string {
 
 // Lowest-level method that reads a document from the bucket.
 func (db *DatabaseContext) GetDoc(docid string) (doc *document, err error) {
+	return db.GetDocument(docid, DocUnmarshalAll)
+}
+
+func (db *DatabaseContext) GetDocument(docid string, unmarshalTo DocumentUnmarshalLevel) (doc *document, err error) {
```

Comment:
It's a bit confusing to have a `GetDoc()` and `GetDocument()`.  What about renaming to to:

`GetFullDocument()` and `GetPartialDocument()`
`GetFullDoc()` and `GetPartialDoc()`  
`GetDoc()` and `GetPartialDoc()`  (where the "full" part is implicit)

Reply:
I agree it's a bit confusing.  GetDocument() is intended to be the standard usage - GetDoc is really just there for backwards compatibility.  

I could look at removing GetDoc altogether (and just have callers pass DocUnmarshalAll to GetDocument), but there are a lot of test usages and leaving the previous GetDoc as a helper method seemed like a reasonable compromise to avoid a lot of refactoring.  If you think it's too confusing, I can bite the bullet and do that refactoring.